desktop/api/src/main/java/org/datacleaner/util/WidgetFactory.java

```diff
@@ -323,6 +333,41 @@ public static JXTextField createTextField(String promptText, int columns) {
         return tf;
     }
 
+    public static JXFormattedTextField createFormattedTextField(String promptText, final int columns, final Format format) {
+        JXFormattedTextField tf = new JXFormattedTextField(promptText);
+        // Stupid JXFormattedTextField will not pass along a formatter to the constructor.
```

Comment:
I don't think so. It is nicely contained within the WidgetFactory, so the pain has been taken care of the the end user.